### DISCUSSION TITLE
cut selections

### DISCUSSION THREAD

**Discussion by fotonfoton123** [MadAnalysis/madanalysis5#241] [General]
Hi,

I have a question related to MadAnalysis. I examine the process p p > z z , z > l-l+, z > vl vl~ through MG5. In this paper, I want to make a better separation of the processes of standard model and new physics beyond the SM with cuts. For this purpose, I write the cuts of the signals and standard model cross sections with the following code. 

#set main.stacking_method = normalize2one
set main.stacking_method = superimpose
#set main.stacking_method = stack

#define vl = 12 14 16 -16 -14 -12 
define invisible = ve ve vm vm vt vt vl vl
#reject PT(a) < 200
set main.graphic_render = root
plot PT(l[1]) 10 0 1000 [logY]
plot PT(l[2]) 10 0 1000 [logY]
plot ETA(l[1]) 1 -5 5 [logY]
plot ETA(l[2]) 1 -5 5 [logY]
plot PT(l[1] l[2]) 50 0 500 [logY]
plot MET 50 0 500 [logY]
plot M(l+ l-) 10 0 1500 [logY]
plot DELTAR(l+ l-) 0.5 0 5 [logY]

As seen in this code, I want to distinguish the signals from the SM from the leptons and neutrinos coming from the z boson. In this code, I can see only ET (missin energy) and M (l-,l+) graphs as output. DeltaR (l-,l+) and ETA (l1), ETA (l2), pt (l1) and pt (l2) graphs are not visible in the plots. I am sending you the code below, I would be glad if you can help. 

Cheers.

**Comment by jackaraz** [comment]
Hi @fotonfoton123, can you give a bit more detail, e.g. version of ma5 that you are using, if you are getting any errors, what ma5 tells you when you run in debug mode, etc

**Comment by fotonfoton123** [comment]
Thank you for your response.  I use to calculate the process p p > z z , z > l-l+, z > vl vl~ through MG5_v3_3_1.  Here I put the signal and SM arkplan lhe files I obtained into madanalysis. Then I open the terminal here and 
./bin/ma5 ppzz.txt  run ma5 with command. ppzz.txt code is as follows:
set main.lumi=500
import /home/mk/madanalysis5/cbb.lhe.gz as cbb

import /home/mk/madanalysis5/cbtw.lhe.gz as cbtw
#
import /home/mk/madanalysis5/cbw.lhe.gz as cbw
#
import /home/mk/madanalysis5/cww.lhe.gz as cww

import /home/mk/madanalysis5/wwsm.lhe.gz as WW

import /home/mk/madanalysis5/zzsm.lhe.gz as ZZ

set cbb.type=signal
set cbb.xsection =0.1092
set cbb.linecolor = red
set cbb.linestyle = dotted
set cbb.linewidth = 2
set cbtw.type=signal
set cbtw.xsection =0.1425
set cbtw.linecolor = gray
set cbtw.linestyle = dotted
set cbtw.linewidth = 2
set cbw.type=signal
set cbw.xsection =0.01843
set cbw.linecolor = green
set cbw.linestyle = dotted
set cbw.linewidth = 2
set cww.type=signal
set cww.xsection =0.01524 
set cww.linecolor = black
set cww.linestyle = dotted
set cww.linewidth = 2
set wwsm.type=background
set wwsm.xsection =0.02214
set wwsm.linecolor = pink
set zzsm.linewidth = 2
set zzsm.type=background
set zzsm.xsection =0.000703 
set zzsm.linecolor = yellow
set zzsm.linewidth = 2


#set main.stacking_method = normalize2one
set main.stacking_method = superimpose
#set main.stacking_method = stack

#define vl = 12 14 16 -16 -14 -12 
define invisible = ve ve vm vm vt vt vl vl
#reject PT(a) < 200
set main.graphic_render = root
plot PT(l[1]) 10 0 1000 [logY]
plot PT(l[2]) 10 0 1000 [logY]
plot ETA(l[1]) 1 -5 5 [logY]
plot ETA(l[2]) 1 -5 5 [logY]
plot PT(l[1] l[2]) 50 0 500 [logY]
plot MET 50 0 500 [logY]
plot M(l+ l-) 10 0 1500 [logY]
plot DELTAR(l+ l-) 0.5 0 5 [logY]

submit /home/mk/madanalysis5/results
submit
open

MadAnalysis 5 report draws  only ET (missing energy) and M (l-,l+) graphs as output. 

I am a bit new to the programme, so I hope I have written what you want.

Best regards.

**Reply by jackaraz** [reply]
Hi @fotonfoton123, I need information about Ma5 version that you are using and error/debug messages that ma5 gives you in debug mode.

**Comment by fotonfoton123** [comment]
I use MadAnalysis version 1.9.60 (2021/12/13). After running the code in the terminal, it did not give an error as far as I understand:

ma5>set cbb.type=signal
MA5: ma5>set cbb.xsection =0.1092
MA5: ma5>set cbb.linecolor = red
MA5: ma5>set cbb.linestyle = dotted
MA5: ma5>set cbb.linewidth = 2
MA5: ma5>set cbtw.type=signal
MA5: ma5>set cbtw.xsection =0.1425
MA5: ma5>set cbtw.linecolor = gray
MA5-ERROR: the possible values for the attribute 'linecolor' are 'auto', 'none', 'black', 'white', 'red', 'yellow','blue', 'grey', 'purple', 'cyan', 'orange'.
MA5: ma5>set cbtw.linestyle = dotted
MA5: ma5>set cbtw.linewidth = 2
MA5: ma5>set cbw.type=signal
MA5: ma5>set cbw.xsection =0.01843
MA5: ma5>set cbw.linecolor = green
MA5: ma5>set cbw.linestyle = dotted
MA5: ma5>set cbw.linewidth = 2
MA5: ma5>set cww.type=signal
MA5: ma5>set cww.xsection =0.01524
MA5: ma5>set cww.linecolor = black
MA5: ma5>set cww.linestyle = dotted
MA5: ma5>set cww.linewidth = 2
MA5: ma5>set wwsm.type=background
MA5-ERROR: no object called 'wwsm' is found
MA5: ma5>set wwsm.xsection =0.02214
MA5-ERROR: no object called 'wwsm' is found
MA5: ma5>set wwsm.linecolor = pink
MA5-ERROR: no object called 'wwsm' is found
MA5: ma5>set zzsm.linewidth = 2
MA5-ERROR: no object called 'zzsm' is found
MA5: ma5>set zzsm.type=background
MA5-ERROR: no object called 'zzsm' is found
MA5: ma5>set zzsm.xsection =0.000703
MA5-ERROR: no object called 'zzsm' is found
MA5: ma5>set zzsm.linecolor = yellow
MA5-ERROR: no object called 'zzsm' is found
MA5: ma5>set zzsm.linewidth = 2
MA5-ERROR: no object called 'zzsm' is found
MA5: ma5>#set main.stacking_method = normalize2one
MA5: ma5>set main.stacking_method = superimpose
MA5: ma5>#set main.stacking_method = stack
MA5: ma5>#define vl = 12 14 16 -16 -14 -12
MA5: ma5>define invisible = ve ve vm vm vt vt vl vl
MA5-WARNING: Particle/Multiparticle labelled 'invisible' is already defined.
MA5-WARNING: Would you like to overwrite the previous definition ? (Y/N)
Answer: y
MA5: ma5>#reject PT(a) < 200
MA5: ma5>set main.graphic_render = root
MA5: ma5>plot PT(l[1]) 10 0 1000 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot PT(l[2]) 10 0 1000 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot ETA(l[1]) 1 -5 5 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot ETA(l[2]) 1 -5 5 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot PT(l[1] l[2]) 50 0 500 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot MET 50 0 500 [logY]
MA5: ma5>plot M(l+ l-) 10 0 1500 [logY]
MA5: ma5>plot DELTAR(l+ l-) 0.5 0 5 [logY]
MA5-ERROR: argument nr. 6(nbins) must have a non-zero, positive, integer value.
MA5: ma5>submit /home/mk/madanalysis5/results
MA5:    Creating folder 'results'..

**Comment by fotonfoton123** [comment]
WARNING: Particle/Multiparticle labelled 'invisible' is already defined.
MA5-WARNING: Would you like to overwrite the previous definition ? (Y/N)
Answer: y
MA5: ma5>#reject PT(a) < 200
MA5: ma5>set main.graphic_render = root
MA5: ma5>plot PT(l[1]) 10 0 1000 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot PT(l[2]) 10 0 1000 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot ETA(l[1]) 1 -5 5 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot ETA(l[2]) 1 -5 5 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot PT(l[1] l[2]) 50 0 500 [logY]
MA5-ERROR: 'l' is not a defined (multi)particle.
MA5: ma5>plot MET 50 0 500 [logY]
MA5: ma5>plot M(l+ l-) 10 0 1500 [logY]
MA5: ma5>plot DELTAR(l+ l-) 0.5 0 5 [logY]
MA5-ERROR: argument nr. 6(nbins) must have a non-zero, positive, integer value.
MA5: ma5>submit /home/mk/madanalysis5/results
MA5:    Creating folder 'results'...
MA5-WARNING: A directory called '/home/mk/madanalysis5/results' is already defined.
Would you like to remove it ? (Y/N)
Answer: y

**Reply by jackaraz** [reply]
Okay, so as you can see from all these error messages, you haven't defined the particle ``l``, and your ``DELTAR`` definition is wrong. The convention is ``plot DELTAR(...) <number of bins> <min x val> <max x val> [arguments]``. Once you fix these, it will produce your plots. FYI, your MA5 version is pretty old

**Comment by fotonfoton123** [comment]
Hi, 
I understand the solution to my problems. However, I have one more question. Is there a document or any other resource you can recommend for all the cuts used in Madanalysis and the graphs to be drawn for them?

Cheers.

**Reply by jackaraz** [reply]
You can find all the documentation in [here](https://github.com/MadAnalysis/madanalysis5/blob/main/CITATIONS.bib) and additionally some partial information [here](http://madanalysis.irmp.ucl.ac.be/).

**Comment by fotonfoton123** [comment]
thank you for your reply.

Cheers.
